### DISCUSSION TITLE
在使用suburl模式时候碰到的问题

### DISCUSSION THREAD

**Discussion by hsj1992** [kkkgo/PaoPaoGateWay#22] [问和答] [answered]
1.如果设置的suburl链接无法直连的话，
一开始选择suburl模式，虚拟机启动后会因此无法正常工作。

2.而当它失败后，尝试再切yaml模式使用本地的custom.yaml，也会失败。

3.所以只能先设置yaml模式，启动虚拟机，使用custom.yaml正常使用clash后，再切suburl模式，才不会出问题。

但是ini文件放在了suburl模式，一旦虚拟机重启，又会回到1的情况。

**Comment by kkkgo** [comment]
理论上，下载订阅配置是不会走代理的，切换模式不会影响到下载效果。  
详细问题，贴出ppgw.ini的配置，以及运行日志截图才能分析你遇到的问题。

**Comment by kkkgo** [comment]
>1.如果设置的suburl链接无法直连的话，
一开始选择suburl模式，虚拟机启动后会因此无法正常工作。

这是当然，你提供的订阅无法下载，自然无法工作。网关会循环尝试直到能下载订阅成功为止，或者检测到ini被修改后重新加载配置。


>2.而当它失败后，尝试再切yaml模式使用本地的custom.yaml，也会失败。

未复现，请确保使用的是最新版本，以及贴出ppgw.ini配置。

>3.所以只能先设置yaml模式，启动虚拟机，使用custom.yaml正常使用clash后，再切suburl模式，才不会出问题。

如上所述，切换模式不会对下载订阅造成影响。下载订阅永远都是直连，网关不会对自身的流量进行任何处理。因此你遇到的问题可能是其他方面的。你可以截图虚拟机运行日志来分析。

**Reply by hsj1992** [reply, marked as answer]
虽然没能再次复现，但我重新查看了您写的说明，确认了我的问题：
“比如RULE-SET功能在启动的时候下载远程url文件失败的话可能会导致clash无法正常启动，而clash无法启动的时候文件可能不能被正常下载，进入了死循环。”
确实是当时网络不稳定，下载的文件内容不完整且损坏导致的。

PaoPaoGateWay和PaoPaoDNS都很好用，真是太感谢您了！
